src/main/java/frc/robot/subsystems/Lights.java

```diff
@@ -3,27 +3,74 @@
 import edu.wpi.first.wpilibj.AddressableLED;
 import edu.wpi.first.wpilibj.AddressableLEDBuffer;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
 import frc.robot.RobotMap;
+import java.util.TreeMap;
 
 /** Controls the LEDs that indicate robot status */
 public class Lights extends SubsystemBase {
   private LightCode currentLightStatus = LightCode.OFF;
+  private TreeMap<LightCode, AddressableLEDBuffer> lightOptionsMap;
   private AddressableLED led = new AddressableLED(RobotMap.LED_PWM_PORT);
-  private AddressableLEDBuffer ledBuffer;
+  private AddressableLEDBuffer ledBufferCone;
+  private AddressableLEDBuffer ledBufferCube;
+  private AddressableLEDBuffer ledBufferGameObject;
+  private AddressableLEDBuffer ledBufferNoTag;
+  private AddressableLEDBuffer ledBufferWorking;
+  private AddressableLEDBuffer ledBufferReadyToScore;
+  private AddressableLEDBuffer ledBufferOff;
 
   /** for example, a spacing value of 1 would mean every LED is on */
   private int spacing = 1;
 
-  /** timer for blinking led */
+  /** timer for blinking LED */
   private int blinkingTimer = 0;
   /** When the blinkingTimer reaches 10 (equal to blinkingPeriod), the LED is toggled */
   private int blinkingPeriod = 10;
-  /** True when light status should be blinking (ex. NO_TAG) */
-  private boolean currentlyBlinking = false;
+  /** If we are currently blinking, then True would mean the color is currently showing */
+  private boolean blinkingColorOn = false;
 
   public Lights() {
-    ledBuffer = new AddressableLEDBuffer(60);
-    led.setLength(ledBuffer.getLength());
+    ledBufferOff = new AddressableLEDBuffer(Constants.LED_LENGTH);
+    led.setLength(ledBufferOff.getLength());
+
+    // sets the rgb values for each of the AddressableLEDBuffers
+    for (int i = 0; i < ledBufferCone.getLength(); i = i + spacing) {
+      ledBufferCone.setRGB(i, 255, 255, 0);
+    }
```

Comment:
Can you add a helper function that fills a given buffer with provided RGB values so that we don't have a bazillion for loops